DataFormats/Detectors/PHOS/src/EventHandler.cxx

```diff
@@ -0,0 +1,215 @@
+    } else {
+      throw InteractionRecordInvalidException(irClusters.value(), irCells.value());
+    }
+  } else if (irClusters) {
+    return irClusters.value();
+  } else if (irCells) {
+    return irCells.value();
+  }
+  throw NotInitializedException();
+}
+
+template <class CellInputType>
+const typename EventHandler<CellInputType>::ClusterRange EventHandler<CellInputType>::getClustersForEvent(int eventID) const
+{
+  if (mTriggerRecordsClusters.size()) {
+    if (eventID >= mTriggerRecordsClusters.size()) {
+      throw RangeException(eventID, mTriggerRecordsClusters.size());
+    }
+    auto& trgrecord = mTriggerRecordsClusters[eventID];
+    return ClusterRange(mClusters.data() + trgrecord.getFirstEntry(), trgrecord.getNumberOfObjects());
+  }
+  throw NotInitializedException();
+}
+
+template <class CellInputType>
+const typename EventHandler<CellInputType>::CellRange EventHandler<CellInputType>::getCellsForEvent(int eventID) const
+{
+  if (mTriggerRecordsCells.size()) {
+    if (eventID >= mTriggerRecordsCells.size()) {
+      throw RangeException(eventID, mTriggerRecordsCells.size());
+    }
+    auto& trgrecord = mTriggerRecordsCells[eventID];
+    return CellRange(mCells.data() + trgrecord.getFirstEntry(), trgrecord.getNumberOfObjects());
+  }
+  throw NotInitializedException();
+}
+
+template <class CellInputType>
+const typename EventHandler<CellInputType>::CellIndexRange EventHandler<CellInputType>::getClusterCellIndicesForEvent(int eventID) const
+{
+  if (mTriggerRecordsCellIndices.size()) {
+    if (eventID >= mTriggerRecordsCellIndices.size()) {
+      throw RangeException(eventID, mTriggerRecordsCellIndices.size());
+    }
+    auto& trgrecord = mTriggerRecordsCellIndices[eventID];
+    return CellIndexRange(mClusterCellIndices.data() + trgrecord.getFirstEntry(), trgrecord.getNumberOfObjects());
+  }
+  throw NotInitializedException();
+}
+
+template <class CellInputType>
+void EventHandler<CellInputType>::reset()
+{
+  mTriggerRecordsClusters = TriggerRange();
+  mTriggerRecordsCellIndices = TriggerRange();
+  mTriggerRecordsCells = TriggerRange();
+  mClusters = ClusterRange();
+  mClusterCellIndices = CellIndexRange();
+  mCells = CellRange();
+}
+
+template <class CellInputType>
+EventData<CellInputType> EventHandler<CellInputType>::buildEvent(int eventID) const
+{
+  EventData<CellInputType> outputEvent;
+  outputEvent.mInteractionRecord = getInteractionRecordForEvent(eventID);
+  if (hasClusters()) {
+    outputEvent.mClusters = getClustersForEvent(eventID);
+  }
+  if (hasClusterIndices()) {
+    outputEvent.mCellIndices = getClusterCellIndicesForEvent(eventID);
+  }
+  if (hasCells()) {
+    outputEvent.mCells = getCellsForEvent(eventID);
+  }
+
+  return outputEvent;
+}
+
+template <class CellInputType>
+bool EventHandler<CellInputType>::compareInteractionRecords(const InteractionRecord& lhs, const InteractionRecord& rhs) const
+{
+  return lhs.bc == rhs.bc && lhs.orbit == rhs.orbit;
+}
```

Comment:
just for the record: InteractionRecord has operator==, so one can do directly `lhs==rhs`